src/lib.rs

```diff
@@ -5,6 +5,11 @@ pub use builder::Builder;
 mod config;
 use config::Config;
 
+mod connection;
+pub use connection::Connection;
+pub use connection::ConnectionIo;
+pub use connection::Segment;
```

Comment:
Not a blocker, but `pub use connection::{Connection, Connectionio, Segment};` is more idiomatic here.l

Reply:
sg